File: azan_local_notifications/android/src/main/java/com/ahmed/azan_local_notifications/ForegroundServiceStartParameter.java
package com.ahmed.azan_local_notifications;

import com.ahmed.azan_local_notifications.models.NotificationDetails;

import java.io.Serializable;
import java.util.ArrayList;

public class ForegroundServiceStartParameter implements Serializable {
  public static final String EXTRA =
      "com.ahmed.azan_local_notifications.ForegroundServiceStartParameter";

  public final NotificationDetails notificationData;
  public final int startMode;
  public final ArrayList<Integer> foregroundServiceTypes;

  public ForegroundServiceStartParameter(
      NotificationDetails notificationData,
      int startMode,
      ArrayList<Integer> foregroundServiceTypes) {
    this.notificationData = notificationData;
    this.startMode = startMode;
    this.foregroundServiceTypes = foregroundServiceTypes;
  }

  @Override
  public String toString() {
    return "ForegroundServiceStartParameter{"
        + "notificationData="
        + notificationData
        + ", startMode="
        + startMode
        + ", foregroundServiceTypes="
        + foregroundServiceTypes
        + '}';
  }
}

File: azan_local_notifications/android/src/main/kotlin/com/ahmed/azanlocalnotifications/ForegroundServiceStartParameter.kt
package com.ahmed.azanlocalnotifications

import com.dexterous.flutterlocalnotifications.models.NotificationDetails
import java.io.Serializable
import java.util.ArrayList

class ForegroundServiceStartParameter(
    notificationData: NotificationDetails,
    startMode: Int,
    foregroundServiceTypes: ArrayList<Integer?>
) : Serializable {
    val notificationData: NotificationDetails
    val startMode: Int
    val foregroundServiceTypes: ArrayList<Integer>

    @Override
    override fun toString(): String {
        return ("ForegroundServiceStartParameter{"
                + "notificationData="
                + notificationData
                + ", startMode="
                + startMode
                + ", foregroundServiceTypes="
                + foregroundServiceTypes
                + '}')
    }

    companion object {
        const val EXTRA = "com.dexterous.flutterlocalnotifications.ForegroundServiceStartParameter"
    }

    init {
        this.notificationData = notificationData
        this.startMode = startMode
        this.foregroundServiceTypes = foregroundServiceTypes
    }
}